Framework/Algorithms/src/PolarizationCorrections/PolarizerEfficiency.cpp

```diff
@@ -0,0 +1,201 @@
+    errorList[PropertyNames::OUTPUT_WORKSPACE] = "Either an output workspace or output file must be provided.";
+  }
+
+  return errorList;
+}
+
+void PolarizerEfficiency::exec() { calculatePolarizerEfficiency(); }
+
+void PolarizerEfficiency::calculatePolarizerEfficiency() {
+  // First we extract the individual workspaces corresponding to each spin configuration from the group workspace
+  const WorkspaceGroup_sptr &groupWorkspace = getProperty(PropertyNames::INPUT_WORKSPACE);
+  const auto spinConfigurationInput = getPropertyValue(PropertyNames::SPIN_STATES);
+
+  const auto &t01Ws = PolarizationCorrectionsHelpers::workspaceForSpinState(groupWorkspace, spinConfigurationInput,
+                                                                            SpinStateValidator::ZERO_ONE);
+  const auto &t00Ws = PolarizationCorrectionsHelpers::workspaceForSpinState(groupWorkspace, spinConfigurationInput,
+                                                                            SpinStateValidator::ZERO_ZERO);
+
+  auto effCell = convertToHistIfNecessary(getProperty(PropertyNames::ANALYSER_EFFICIENCY));
+
+  auto rebin = createChildAlgorithm("RebinToWorkspace");
+  rebin->initialize();
+  rebin->setProperty("WorkspaceToRebin", effCell);
+  rebin->setProperty("WorkspaceToMatch", t00Ws);
+  rebin->setProperty("OutputWorkspace", "rebinToWorkspace");
+  rebin->execute();
+  effCell = rebin->getProperty("OutputWorkspace");
+
+  auto &&effPolarizer = (t00Ws - t01Ws) / (4 * (2 * effCell - 1) * (t00Ws + t01Ws)) + 0.5;
+
+  calculateErrors(t00Ws, t01Ws, effCell, effPolarizer);
+
+  auto const &filename = getPropertyValue(PropertyNames::OUTPUT_FILE_PATH);
+  if (!filename.empty()) {
+    saveToFile(effPolarizer, filename);
+  }
+
+  auto const &outputWsName = getPropertyValue(PropertyNames::OUTPUT_WORKSPACE);
+  if (!outputWsName.empty()) {
+    setProperty(PropertyNames::OUTPUT_WORKSPACE, effPolarizer);
+  }
+}
+
+void PolarizerEfficiency::calculateErrors(const MatrixWorkspace_sptr &t00Ws, const MatrixWorkspace_sptr &t01Ws,
+                                          const MatrixWorkspace_sptr &effCellWs,
+                                          const MatrixWorkspace_sptr &effPolarizerWs) {
+
+  auto &effPolarizerE = effPolarizerWs->mutableE(0);
+  const auto &t00E = t00Ws->e(0);
+  const auto &t01E = t01Ws->e(0);
+  const auto &effCellE = effCellWs->e(0);
+  const auto &t00Y = t00Ws->y(0);
+  const auto &t01Y = t01Ws->y(0);
+  const auto &effCellY = effCellWs->y(0);
+
+  for (size_t i = 0; i < effPolarizerE.size(); ++i) {
+    const auto &delta00 = (t01Y[i]) / (2 * (2 * effCellY[i] - 1) * pow(t00Y[i] + t01Y[i], 2));
+    const auto &delta01 = (-1 * t00Y[i]) / (2 * (2 * effCellY[i] - 1) * pow(t00Y[i] + t01Y[i], 2));
+    const auto &deltaEffCell = (t01Y[i] - t00Y[i]) / (pow(2 * (2 * effCellY[i] - 1), 2) * (t00Y[i] + t01Y[i]));
+    effPolarizerE[i] = sqrt(pow(abs(delta00), 2) * pow(t00E[i], 2) + pow(abs(delta01), 2) * pow(t01E[i], 2) +
```

Comment:
Yes no need for `abs` as you say. Also I thought `pow(num, 2)` was slower than `num * num` - but with equations this long that might make them difficult to read! I think you could at least call `pow` fewer times e.g. 
```suggestion
    effPolarizerE[i] = sqrt(pow(delta00*t00E[i], 2) + pow(delta01*t01E[i], 2) +
```
It does look like some of the terms are computed in multiple places (e.g. `(2 * effCellY[i] - 1)` and `t00Y[i] + t01Y[i]`) - I don't know whether it's worth putting them in variables to make this obvious (I don't know whether that actually improves speed, I guess not as there's some overhead to assigning the variable). It might be more readable?

Reply:
Thanks for this Richard, some useful points that I'll take a look at. I hadn't realised (remembered? - maybe at some point I would have come across this a long time ago 🙂) that mathematically $x^2 * y^2$ is equivalent to $(x*y)^2$, so that's really useful to know.